scripts/common.js

```diff
@@ -647,8 +647,185 @@ export function extractTopicsAndProducts() {
   }
 }
 
+/**
+ * Set up click event on Region Picker Feds
+ * Builds Dropdown Based on Region List
+ * Get selected region from Dropdown Selector
+ */
+function handleDropdownRegion() {
+  const regionsNameList = [
+    {
+      lang: 'en_apac',
+      name: 'APAC (English)',
+      home: `/en/apac.html`,
+    },
+    {
+      lang: "ko",
+      name: "Korea (한국어)",
+      home: `/ko/`,
+    },
+    {
+      lang: "en_uk",
+      name: "UK (English)",
+      home: `/en/uk.html`,
+    },
+    {
+      lang: "en",
+      name: "US (English)",
+      home: `/`,
+    }
+  ];
+
+  // Add Region Dropdown Container before Feds Footer
+  const fedsFooter = document.querySelector('#feds-footer');
+  if (fedsFooter) {
+    const regionDropdownContainer = document.createElement('div');
+    regionDropdownContainer.classList.add('region-dropdown');
+    regionDropdownContainer.innerHTML = `<ul class="region-dropdown-list"></ul>`;
+    fedsFooter.parentElement.insertBefore(regionDropdownContainer, fedsFooter);
+  }
+  
+  // Get actual selected Region
+  const getSelectedRegion = () => {
+    const regionPage = regionsNameList.find(r => r.home === location.pathname);
+    // check if Region name matches the blog.language example 'en' in order to show the Region Name in the buttom
+    const langPage = regionsNameList.find(r => r.lang === window.blog.language);
+    let regionLang;
+    let regionName;
+    if (!regionPage) {
+      // region array is empty, we are not on a region page -> check the sessionStorage, if no value, use blog.language
+      regionLang = sessionStorage.getItem('blog-selected-language') || window.blog.language;
```

Comment:
I think some refactoring is needed to make actions clear:
- by default you get the language (and name maybe) from session
- if that's not available, you identify the language based on `blog.language` property; there's a special case for `en`.
- when the user clicks on a different region, a redirect happens and the URL (pathname) matches the `home` property: this is when you can update your session storage value

Reply:
having this input as enhancement for the code